api-2.2/src/main/java/org/openmrs/htmlformentryui/ConditionElement.java

```diff
@@ -0,0 +1,309 @@
+				}	
+		    } else {
+		    	ret.add(new FormSubmissionError(context
+	                    .getFieldName(conditionStatusWidget), "A Condition status is required"));
+		    }
+		}
+		return ret;
+	}
+
+	@Override
+	public String generateHtml(FormEntryContext context) {
+		mss = Context.getMessageSourceService();		
+		StringBuilder ret = new StringBuilder();
+		ret.append("<h5>");
+		ret.append(mss.getMessage("coreapps.conditionui.addNewCondition"));
+		ret.append("</h5>");
+		ret.append(createConditionNameWidget(context));
+		ret.append(createConditionDateWidgets(context));
+		ret.append(createConditionStatusWidgets(context));		
+		return ret.toString();
+	}
+		
+	public String createConditionNameWidget(FormEntryContext context) {
+		List<ConceptClass> requiredClasses = new ArrayList<ConceptClass>();
+		Set<Concept> initialConcepts = new HashSet<Concept>();
+		if (mss == null) {
+			mss = Context.getMessageSourceService();		
+		}
+		
+		ConceptClass classCache = Context.getConceptService().getConceptClassByUuid("8d4918b0-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		classCache = Context.getConceptService().getConceptClassByUuid("8d492b2a-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		classCache = Context.getConceptService().getConceptClassByUuid("8d491a9a-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		
+		for (ConceptClass cc : requiredClasses) {
+			initialConcepts.addAll(Context.getConceptService().getConceptsByClass(cc));
+		}
+		
+		conditionName = new ConceptSearchAutocompleteWidget(new ArrayList<Concept>(initialConcepts), requiredClasses);
+		String conditionNameTextInputId = context.registerWidget(conditionName);
+		conditionNameErrorWidget = new ErrorWidget();
+		context.registerErrorWidget(conditionName, conditionNameErrorWidget);
+		
+		StringBuilder ret = new StringBuilder();
+		ret.append(conditionName.generateHtml(context));
+		if (context.getMode() != Mode.VIEW) {
+			ret.append(conditionNameErrorWidget.generateHtml(context));
+		}
+		ret.append("\n<script>jq('#" + conditionNameTextInputId + "').attr('placeholder',");
+		ret.append(" '" + mss.getMessage("coreapps.conditionui.condition") + "');\n");
+		// Add support for non-coded concept values.
+		// This a hack to let the autocomplete widget accept values that aren't part of the concept list.
+		ret.append("jq('#" + conditionNameTextInputId + "').blur(function(e){\n");
+		ret.append("     var valueAttr = jq('#" + conditionNameTextInputId + "_hid" + "').attr('value');\n");
+		ret.append("     if(valueAttr === \"ERROR\"){\n");
+		ret.append("        jq('#" + conditionNameTextInputId + "_hid" + "').attr('value', '');\n");
+		ret.append("     }\n");
+		ret.append("});\n");				
+		ret.append("</script>\n");
+		return ret.toString();
+	}
+	
+	private String createConditionStatusWidgets(FormEntryContext context) {
```

Comment:
I would rather see a `protected` method named `htmlForConditionStatusWidgets(FormEntryContext)`.

Reply:
Why don't we have it `private` ?